### PR TITLE
[WIP] [MachineLearning.Nnstreamer] Add Nnstreamer APIs for Machine Learning

### PR DESCRIPTION
**Pull request by again4you**
### Description of Change ###
This PR initially adds the C# Pipeline APIs for NNStreamer and its test cases.

### Related Link
* Tizen C API ACR: http://suprem.sec.samsung.net/jira/browse/ACR-1382
* NNStreamer: https://github.com/nnsuite/nnstreamer

### Test Result
nnstreamer_testcase

### API Changes ###

 - ACR:

### CODE REVIEW

**Comment by again4you** [comment]
@WonyoungChoi 
This is my first PR draft. Plz go over it and give a comment to fix it.

Thanks in advance.

File: src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    /// <remarks>
+    /// The Pipeline API creates a stream pipeline with NNStreamer plugins, GStreamer plugins, and sensor input devices.
+    /// It provides the interface to push data to the pipeline from the application get the inference result from the pipeline to the application.
+    /// It also provides the Interfaces to start/stop/destroy the pipeline and other utility functions such as controling switches and valves in the pipeline.
+    /// </remarks>
+    public class Pipeline
+    {
+        static string Tag = "Nnstreamer.Pipeline";
```
Comment (WonyoungChoi) [inline src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs]: use `const` or `static readonly`

File: src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        /// API to construct the pipeline (GStreamer + NNStreamer)
+        /// </summary>
+        /// <param name="pipelineDesc">
+        /// The pipeline description compatible with GStreamer gst_parse_launch(). 
+        /// Refer to GStreamer manual or NNStreamer (github.com/nnsuite/nnstreamer) documentation for examples and the grammar.
+        /// </param>
+        /// <exception cref="InvalidParameter">Invalid Parameter</exception>
+        /// <exception cref="IOException">Pipeline construction is failed because of wrong parameter or initialization failure.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public void Construct(string pipelineDesc)
```
Comment (WonyoungChoi) [inline src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs]: Why don't you use the constructor and finalizer of the Pipeline class?

File: src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    /// <since_tizen> 6 </since_tizen>
+    /// <remarks>
+    /// The Pipeline API creates a stream pipeline with NNStreamer plugins, GStreamer plugins, and sensor input devices.
+    /// It provides the interface to push data to the pipeline from the application get the inference result from the pipeline to the application.
+    /// It also provides the Interfaces to start/stop/destroy the pipeline and other utility functions such as controling switches and valves in the pipeline.
+    /// </remarks>
+    public class Pipeline
+    {
+        static string Tag = "Nnstreamer.Pipeline";
+        private IntPtr _pipelineHandle;
```
Comment (WonyoungChoi) [inline src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs]: When use native handle as a member of the class, the Disposable pattern should be used for safety. 
see: https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

File: src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            return;
+        }
+
+        /// <summary>
+        /// API to destroy the pipeline
+        /// </summary>
+        /// <exception cref="InvalidParameter">Invalid Parameter</exception>
+        /// <exception cref="IOException">Pipeline destruction is failed because of wrong parameter or initialization failure.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public void Destroy()
```
Comment (WonyoungChoi) [inline src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Pipeline.cs]: It can be moved to the finalizer of the Pipeline class also.

**Comment by myungjoo** [comment]
As we have discussed yesterday, let's have a private constructor (Pipeline) and let a static function to generate a Pipeline instance so that you can have proper Exception.

**Pull request closed without merging**